src/main/java/com/sequenceiq/cloudbreak/service/stack/connector/aws/AwsConnector.java

```diff
@@ -175,31 +139,40 @@ public void buildStack(Stack stack, String userData, Map<String, Object> setupPr
         MDCBuilder.buildMdcContext(stack);
         AwsCredential awsCredential = (AwsCredential) stack.getCredential();
         AmazonCloudFormationClient client = awsStackUtil.createCloudFormationClient(Regions.valueOf(stack.getRegion()), awsCredential);
-        String stackName = String.format("%s-%s", stack.getName(), stack.getId());
-
+        String cFStackName = String.format("%s-%s", stack.getName(), stack.getId());
         List<Parameter> parameters = new ArrayList<>(Arrays.asList(
                 new Parameter().withParameterKey("SSHLocation").withParameterValue("0.0.0.0/0"),
                 new Parameter().withParameterKey("CBUserData").withParameterValue(userData),
-                new Parameter().withParameterKey("StackName").withParameterValue(stackName),
+                new Parameter().withParameterKey("StackName").withParameterValue(cFStackName),
                 new Parameter().withParameterKey("StackOwner").withParameterValue(awsCredential.getRoleArn()),
                 new Parameter().withParameterKey("KeyName").withParameterValue(awsCredential.getKeyPairName()),
                 new Parameter().withParameterKey("AMI").withParameterValue(stack.getImage()),
                 new Parameter().withParameterKey("RootDeviceName").withParameterValue(getRootDeviceName(stack, awsCredential))
         ));
         CreateStackRequest createStackRequest = createStackRequest()
-                .withStackName(stackName)
+                .withStackName(cFStackName)
                 .withOnFailure(com.amazonaws.services.cloudformation.model.OnFailure.valueOf(stack.getOnFailureActionAction().name()))
                 .withTemplateBody(cfTemplateBuilder.build("templates/aws-cf-stack.ftl",
                         spotPriceNeeded(stack.getInstanceGroups()), stack.getInstanceGroupsAsList()))
-                .withNotificationARNs((String) setupProperties.get(SnsTopicManager.NOTIFICATION_TOPIC_ARN_KEY))
                 .withParameters(parameters);
         client.createStack(createStackRequest);
+        Set<Resource> resourceSet = new HashSet<>();
+        Resource resource = new Resource(ResourceType.CLOUDFORMATION_STACK, cFStackName, stack, null);
+        resourceSet.add(resource);
+        stack = stackUpdater.addStackResources(stack.getId(), Arrays.asList(resource));
```

Comment:
- stackUpdater could accept a Set instead of a List, so it won't be necessary to manage both of them
- with Google collections the set creation can be changed to: Sets.newHashSet(resource);